src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs

```diff
@@ -49,17 +49,45 @@ public static Configuration GetConfiguration(IPluginCommunication autoRest)
                 autoRest.GetValue<string?>("namespace").GetAwaiter().GetResult(),
                 autoRest.GetValue<string?>("library-name").GetAwaiter().GetResult(),
                 GetRequiredOption<string[]>(autoRest, "shared-source-folders").Select(TrimFileSuffix).ToArray(),
-                autoRest.GetValue<bool?>("save-inputs").GetAwaiter().GetResult() ?? false,
-                autoRest.GetValue<bool?>("azure-arm").GetAwaiter().GetResult() ?? false,
-                autoRest.GetValue<bool?>("public-clients").GetAwaiter().GetResult() ?? false,
-                autoRest.GetValue<bool?>("model-namespace").GetAwaiter().GetResult() ?? true,
-                autoRest.GetValue<bool?>("head-as-boolean").GetAwaiter().GetResult() ?? false,
-                autoRest.GetValue<bool?>("skip-csproj-packagereference").GetAwaiter().GetResult() ?? false,
-                autoRest.GetValue<bool?>("low-level-client").GetAwaiter().GetResult() ?? false,
+                Get(autoRest, "save-inputs"),
+                Get(autoRest, "azure-arm"),
+                Get(autoRest, "public-clients"),
+                Get(autoRest, "model-namespace"),
+                Get(autoRest, "head-as-boolean"),
+                Get(autoRest, "skip-csproj-packagereference"),
+                Get(autoRest, "low-level-client"),
                 MgmtConfiguration.GetConfiguration(autoRest)
             );
         }
 
+        private static bool Get (IPluginCommunication autoRest, string option)
```

Comment:
Can we call it "GetBool" or something similar?

Reply:
Also, you can have default value as a third parameter, that way you won't need separate `GetDefault`